Sources/DVR/SessionDataTask.swift

```diff
@@ -102,12 +102,13 @@ final class SessionDataTask: URLSessionDataTask {
             let persistInteraction = filteredResponse != nil && filteredRequest != nil
             
             if persistInteraction {
-                
-                if let interaction = this.interaction, let resp = filteredResponse?.response, let data = filteredResponse?.data {
-                    this.interaction = Interaction(request: filteredRequest!, response: resp, responseData: data)
-                    this.session.finishTaskWithInteraction(this, interaction: interaction, playback: false)
-                }
-            } else {
+                guard let resp = filteredResponse?.response else { return }
+                let data = filteredResponse?.data
+                this.interaction = Interaction(request: filteredRequest!, response: resp, responseData: data)
+                guard let interaction = this.interaction else { return }
+                this.session.finishTaskWithInteraction(this, interaction: interaction, playback: false)
+            }
```

Comment:
These are largely the same. The only difference is removing the requirement that data be non-nil. I'd suggest this:

```suggestion
                if let interaction = this.interaction, let resp = filteredResponse?.response {
                    this.interaction = Interaction(request: filteredRequest!, response: resp, responseData: filteredResponse?.data)
                    this.session.finishTaskWithInteraction(this, interaction: interaction, playback: false)
            } else {
```

Reply:
That won't work because "If let interaction = this.interaction" makes the statement evaluate to false.
this.interaction is going to be nil until it's set.

Reply:
The point of the change is that this.interaction is set before its guarded.